File: C++/Medium/Accepted/the-number-of-weak-characters-in-the-game_1.cpp
class Solution {
public:
    int numberOfWeakCharacters(vector<vector<int>>& properties) {
        
        sort(properties.begin(), properties.end(), [](vector<int> &a, vector<int> &b) {
            return a[0] == b[0] ? a[1] > b[1] : a[0] < b[0];
        });
        int res = 0, maxDef = 0;
        for (int i = properties.size() - 1; i >= 0; --i) {
            if (properties[i][1] < maxDef) ++res;
            else maxDef = properties[i][1];
        }
        return res;
    }
};

File: Rust/Medium/Accepted/the-number-of-weak-characters-in-the-game_1.rs
impl Solution {
    pub fn number_of_weak_characters(properties: Vec<Vec<i32>>) -> i32 {
        
        let mut properties = properties;
        properties.sort_by(|a, b| b[0].cmp(&a[0]).then(a[1].cmp(&b[1])));
        let mut max_def = 0;
        let mut ans = 0;
        for p in properties {
            if p[1] < max_def {
                ans += 1;
            } else {
                max_def = p[1];
            }
        }
        ans
    }
}